File: tests/cases/attrs.py
from pyjs import js

@js
class A:
    def __init__(self):
        self.text = "alpha"

@js
class B:
    def __init__(self):
        self.a = A()
        self.beta = "beta"

@js
class C:
    def __init__(self):
        self.b = B()
        self.ceta = "ceta"

@js
class D(C):
    def __init__(self):
        super().__init__()
        self.c = C()
        self.deta = "deta"


def main():
    d = D()
    print(d.deta)
    print(d.ceta)
    print(d.c.ceta)
    print(d.c.b.beta)
    print(d.c.b.a.text)
    print(d.b.a.text)

File: tests/cases/attrs.js
// tests.cases.attrs


class A {

    constructor() {
        this.text = 'alpha';
    }
}

class B {

    constructor() {
        this.a = new A();
        this.beta = 'beta';
    }
}

class C {

    constructor() {
        this.b = new B();
        this.ceta = 'ceta';
    }
}

class D extends C {

    constructor() {
        super();
        this.c = new C();
        this.deta = 'deta';
    }
}

function main() {
    var d = new D();
    console.log(d.deta);
    console.log(d.ceta);
    console.log(d.c.ceta);
    console.log(d.c.b.beta);
    console.log(d.c.b.a.text);
    console.log(d.b.a.text);
}
